### PR TITLE
PR8 - Remove AlicaEngine::this from Logger

### PR DESCRIPTION
**Pull request by triccyx**
Remove AlicaEngine::this from Logger and subclass

### CODE REVIEW

**Comment by veerajsk** [comment]
@triccyx this PR includes changes from PR7 & if I change the base of the PR to PR7 it shows 292 file changes. This means the PR is still not correctly rebased on top of PR7. Can you please rebase correctly so that we can review multiple PRs at once?

**Comment by triccyx** [comment]
Very strange ... I will rebase again

**Comment by triccyx** [comment]
Ok rebase done

File: alica_engine/include/engine/Logger.h [hunk 1]
```diff
@@ -20,6 +20,8 @@ class TeamObserver;
 class EntryPoint;
 class AlicaEngine;
```
Comment (christianhelm) [inline alica_engine/include/engine/Logger.h:21]: Remove AlicaEngine declaration

File: alica_engine/include/engine/teammanager/TeamManager.h [hunk 2]
```diff
@@ -24,6 +24,11 @@ class ActiveAgentIdIterator;
 class ActiveAgentView;
 class ActiveAgentIterator;
 struct AgentQuery;
```
Comment (christianhelm) [inline alica_engine/include/engine/teammanager/TeamManager.h]: Remove AlicaEngine declaration

**Review by christianhelm — COMMENTED** [review]
What has changed in the binary files?

File: alica_engine/include/engine/Logger.h (hunk 1 above)
Comment (bjoernschroeder) [inline alica_engine/include/engine/Logger.h:21]: Forward declaration of engine can be removed

File: alica_engine/src/engine/Logger.cpp [hunk 3]
```diff
@@ -26,8 +26,14 @@ namespace alica
 {
 using std::to_string;
 
-Logger::Logger(AlicaEngine* ae)
-        : _ae(ae)
+Logger::Logger(ConfigChangeListener& configChangeListener, const TeamManager& teamManager, const TeamObserver& teamObserver,
+        const PlanRepository& planRepository, const AlicaClock& clock, const std::string& localAgentName)
+        : _configChangeListener(configChangeListener)
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/Logger.cpp]: We only use the configChangeListener in the constructor. We can use the passed instance to call subscribe and reload once and get rid of storing the configChangeListener locally.

File: alica_engine/src/engine/collections/RobotProperties.cpp [hunk 4]
```diff
@@ -6,12 +6,13 @@
 namespace alica
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/collections/RobotProperties.cpp:6]: Engine include can be removed.

File: alica_engine/src/engine/collections/RobotEngineData.cpp [hunk 5]
```diff
@@ -19,8 +19,9 @@ namespace alica
 /**
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/collections/RobotEngineData.cpp:19]: Engine include can be removed

File: alica_engine/src/engine/teammanager/TeamManager.cpp [hunk 6]
```diff
@@ -3,11 +3,13 @@
 #include "engine/AlicaEngine.h"
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/teammanager/TeamManager.cpp:3]: You can remove the engine include and add the following includes instead:

#include "engine/IAlicaCommunication.h"
#include "engine/ConfigChangeListener.h"
#include "engine/PlanRepository.h"

File: alica_engine/src/engine/teammanager/TeamManager.cpp [hunk 7]
```diff
@@ -53,17 +55,27 @@ bool AgentsCache::addAgent(Agent* agent)
...
 }
 
-TeamManager::TeamManager(AlicaEngine* engine, AgentId agentID)
+TeamManager::TeamManager(ConfigChangeListener& configChangeListener, const ModelManager& modelManager, const PlanRepository& planRepository,
+        const IAlicaCommunication& communicator, const AlicaClock& clock, Logger& log, int version, uint64_t masterPlanId, const std::string& localAgentName,
+        AgentId agentID)
         : _localAgent(nullptr)
-        , _localAgentID(agentID)
-        , _engine(engine)
+        , _configChangeListener(configChangeListener)
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/teammanager/TeamManager.cpp:62]: the configChangeListener is only used in the constructor, we can use the passed instance directly and not store it locally.

**Review by bjoernschroeder — CHANGES_REQUESTED** [review]

File: alica_engine/include/engine/teammanager/TeamManager.h (hunk 2 above)
Comment (triccyx) [reply alica_engine/include/engine/teammanager/TeamManager.h]: ok

File: alica_engine/include/engine/Logger.h (hunk 1 above)
Comment (triccyx) [reply alica_engine/include/engine/Logger.h:21]: ok

File: alica_engine/src/engine/Logger.cpp (hunk 3 above)
Comment (triccyx) [reply alica_engine/src/engine/Logger.cpp]: ok

File: alica_engine/include/engine/Logger.h (hunk 1 above)
Comment (triccyx) [reply alica_engine/include/engine/Logger.h:21]: ok

Comment (triccyx) [reply alica_engine/include/engine/Logger.h:21]: I find out that I forget 2 functions, not class methods, that use AlicaEngine. 
```c++
std::string getLogFilename(AlicaEngine* ae, const std::string& file);
std::string getErrLogFilename(AlicaEngine* ae, const std::string& file);
```
I prefer to correct them in PR15 as leftover.
So I would leave AlicaEngine declaration in this PR

Comment (triccyx) [reply alica_engine/include/engine/Logger.h:21]: @christianhelm 
commit in PR15
https://github.com/rapyuta-robotics/alica/commit/7cad0e7f5b0992ec55ef27a008ef65c063741d1c
and
https://github.com/rapyuta-robotics/alica/commit/b1bcf7b9fde8b289c6d172fccdcc3021240a7bc0

**Comment by christianhelm** [comment]
The jar files need to be removed from this PR

**Comment by triccyx** [comment]
> The jar files need to be removed from this PR

The only way I found to remove the files from commit was:
```
git reset --soft <the old commit>
git reset HEAD <file>
git commit --amend
push --force ....
```
Before I have tried simply with 
```
git checkout origin/rr_devel   -- <file name> 
```
but it won't work I think for some lfs issue.

It seems to me that this time the two files disappear from  PR and I hope I haven't 
destroy the branch

**Review by christianhelm — APPROVED** [review]

**Review by veerajsk — APPROVED** [review]

**Review by bjoernschroeder — APPROVED** [review]

**Pull request merged**